packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -165,11 +137,30 @@ export function FlatTermSelector( { slug } ) {
 	}, [ searchResults ] );
 
 	const { editPost } = useDispatch( editorStore );
+	const { saveEntityRecord } = useDispatch( coreStore );
 
 	if ( ! hasAssignAction ) {
 		return null;
 	}
 
+	async function createTerm( term ) {
+		try {
+			const newTerm = await saveEntityRecord( 'taxonomy', slug, term, {
+				throwOnError: true,
+			} );
+			return unescapeTerm( newTerm );
+		} catch ( error ) {
+			if ( error.code !== 'term_exists' ) {
+				throw error;
+			}
+
+			return {
+				id: error.data.term_id,
+				name: term.name,
+			};
```

Comment:
@Mamaduka only to stay true to the original impentation shouldn't we also run the term in the catch statement through the `unescapeTerm` function ?

```suggestion
			return unescapeTerm({
				id: error.data.term_id,
				name: term.name,
			});
```

Reply:
The provided `term.name` isn't "escaped", so I think there's no need for it.

Reply:
Ahh yeah, you are right. I had mixed something up in my head...